docs/2021/microservice/updates/2021-07-13.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Week 6
+author: Omar AbdelSamea
+---
+<!--
+SPDX-License-Identifier: CC-BY-SA-4.0
+
+SPDX-FileCopyrightText: 2021 Omar AbdelSamea <omarmohamed168@gmail.com>
+-->
+
+# Coding Week 6
+
+## Meeting 0
+### Organizer
+- Gaurav Mishra
+
+### Attendees
+- Anupam Ghosh
+- Shaheem Azmal
+- Vasudev Maduri
+- Klaus Gmeinwieser
+- Omar AbdelSamea
+
+### Outcomes
+- Discussed the issue of having agents conf seperating between containers and the idle solution would be to add all agents conf to a database.
+- for database system discussed two options etcd and reddis.
+- Discussed how etcd should be used (seperate high availblity cluster vs deployment) and the discision was using a deployment as the seperate cluster for etcd only will be an overkill
+
+## Meeting 1
+### Organizer
+- Gaurav Mishra
+
+### Attendees
+### Organizer
+- Gaurav Mishra
+
+### Attendees
+- Anupam Ghosh
+- Vasudev
+- Ayush Bharadwaj
+- Shreya Singh
+- Kaushlendra Pratap Singh
+- Omar AbdelSamea
+
+### Outcomes
+- Made a small demo for working FOSSology installation applied with some hacks (agents conf and web UI)
+- Presented manifest of etcd (deployment, service and pvc).
+- Presented a concept of how etcd will be used inside scheduler code.
+
+## Migrating configurations from conf files to etcd
+- For microservice architecture each agent has it's own conf file inside it's pod.
+- FOSSology hosts needs to be hardcoded before deployment.
+- To solve this all conf files will be added to a key value pair database. The selected db system is [etcd](https://etcd.io/)  
+- for each new agent added, it'll interface with etcd using RESTful API. The agent will open it's conf file and start making proper PUT request to add configuration in etcd and the agent will add the host details.
+
+
+## End of the week result 
+- Succesfully scaned files using copyright and nomos (some hacks are applied to get it working)
+- Compared etcd vs redis and at the end etcd will be used
+- Started migrating Hosts part from FOSSology conf to etcd by modfying `scheduler_foss_config` in [scheduler](https://github.com/OmarAbdelSamea/fossology/blob/OmarAbdelSamea/GSoC/Microservices/src/scheduler/agent/scheduler.c#L971)
+
+## Diagrams and screenshots
+### Interface between k8s, scheduler and etcd.
+![image](https://user-images.githubusercontent.com/57943026/128032268-0fb2777b-70f5-4ae7-bc44-0786114fe277.png)
```

Comment:
To enable reliable storage, please upload the image in the repository under `static/img/microservice/` and change the path to `![image](/img/microservice/<image_name>.png)`

Same need to be done for other images.